backend/api/v1/project_service.go

```diff
@@ -266,6 +266,128 @@ func (s *ProjectService) SetIamPolicy(ctx context.Context, request *v1pb.SetIamP
+	trimDeploymentConfigSuffix, err := trimSuffix(request.Name, deploymentConfigSuffix)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+	projectID, err := getProjectID(trimDeploymentConfigSuffix)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+	project, err := s.store.GetProjectV2(ctx, &store.FindProjectMessage{
+		ResourceID: &projectID,
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, err.Error())
+	}
+	if project == nil {
+		return nil, status.Errorf(codes.NotFound, "project %q not found", request.Name)
+	}
+	if project.Deleted {
+		return nil, status.Errorf(codes.InvalidArgument, "project %q has been deleted", request.Name)
+	}
+
+	deploymentConfig, err := s.store.GetDeploymentConfigV2(ctx, project.UID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, err.Error())
+	}
+	if deploymentConfig == nil {
+		return nil, status.Errorf(codes.NotFound, "deployment config %q not found", request.Name)
+	}
+
+	return convertToDeploymentConfig(project.ResourceID, deploymentConfig), nil
+}
+
+// UpdateDeploymentConfig updates the deployment config for a project.
+func (s *ProjectService) UpdateDeploymentConfig(ctx context.Context, request *v1pb.UpdateDeploymentConfigRequest) (*v1pb.DeploymentConfig, error) {
+	if request.Config == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "deployment config is required")
+	}
+	trimDeploymentConfigSuffix, err := trimSuffix(request.Config.Name, deploymentConfigSuffix)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+	projectID, err := getProjectID(trimDeploymentConfigSuffix)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+	project, err := s.store.GetProjectV2(ctx, &store.FindProjectMessage{
+		ResourceID: &projectID,
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, err.Error())
+	}
+	if project == nil {
+		return nil, status.Errorf(codes.NotFound, "project %q not found", projectID)
+	}
+	if project.Deleted {
+		return nil, status.Errorf(codes.InvalidArgument, "project %q has been deleted", projectID)
+	}
+
+	if err := validateSchedule(request.Config.Schedule); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	storeDeploymentConfig, err := convertToStoreDeploymentConfig(request.Config)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	deploymentConfig, err := s.store.UpsertDeploymentConfigV2(ctx, &store.UpsertDeploymentConfigMessage{
+		ProjectUID:       project.UID,
+		PrincipalUID:     ctx.Value(common.PrincipalIDContextKey).(int),
+		DeploymentConfig: storeDeploymentConfig,
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, err.Error())
+	}
+	return convertToDeploymentConfig(project.ResourceID, deploymentConfig), nil
+}
+
+func validateSchedule(schedule *v1pb.Schedule) error {
```

Comment:
Also can we combine validateSchedule and convertToStoreDeploymentConfig which does convertion and validation at the same time?